File: 6/max_average_subarray1.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        pre_sum = sum(nums[0:k])
        max_sum = pre_sum

        for i in range(len(nums)-k):
            next_sum = pre_sum - nums[i] + nums[i + k]
            if next_sum > max_sum:
                max_sum = next_sum
            pre_sum = next_sum

        return max_sum/k
        

File: 6/max_average_subarray1.go
package main

import "fmt"

func sum(nums []int) int {
	var sum int
	for _, v := range nums {
		sum += v
	}
	return sum
}

func findMaxAverage(nums []int, k int) float64 {
	preSum := sum(nums[0:k])
	maxSum := preSum
	var nextSum int
	for i := 0; i < len(nums)-k; i++ {
		nextSum = preSum - nums[i] + nums[i+k]
		if nextSum > maxSum {
			maxSum = nextSum
		}
		preSum = nextSum
	}
	return float64(maxSum) / float64(k)
}

func main() {
	sli := []int{1, 12, -5, -6, 50, 3}
	fmt.Println(findMaxAverage(sli, 4))
}
